platform/broadcom/sonic-platform-modules-ufispace/s9300-32d/sonic_platform/chassis.py

```diff
@@ -0,0 +1,220 @@
+            print(
+                "get_change_event:" "time wrap / invalid timeout value",
+                timeout,
+            )
+            return False, change_event_dict  # Time wrap or possibly incorrect timeout
+        try:
+            while timeout >= 0:
+                # check for sfp
+                sfp_change_dict = self.get_transceiver_change_event()
+
+                if sfp_change_dict:
+                    change_event_dict["sfp"] = sfp_change_dict
+                    return True, change_event_dict
+                if forever:
+                    time.sleep(1)
+                else:
+                    timeout = end_time - time.time()
+                    if timeout >= 1:
+                        time.sleep(1)  # We poll at 1 second granularity
+                    else:
+                        if timeout > 0:
+                            time.sleep(timeout)
+                        return True, change_event_dict
+        except Exception as e:
+            print(e)
+        print("get_change_event: Should not reach here.")
+        return False, change_event_dict
+
+    def get_transceiver_change_event(self, timeout=0):
+        current_port_dict = {}
+        ret_dict = {}
+
+        # Check for OIR events and return ret_dict
+        for index in range(self.platform_inventory['num_ports']):
+            if self._sfp_list[index].get_presence():
+                current_port_dict[index] = self.plugin_data['XCVR']['plug_status']['inserted']
+            else:
+                current_port_dict[index] = self.plugin_data['XCVR']['plug_status']['removed']
+
+        if len(self.port_dict) == 0:       # first time
+            self.port_dict = current_port_dict
+            return {}
+
+        if current_port_dict == self.port_dict:
+            return {}
+
+        # Update reg value
+        for index, status in current_port_dict.items():
+            if self.port_dict[index] != status:
+                ret_dict[index] = status
+                #ret_dict[str(index)] = status
+        self.port_dict = current_port_dict
+        for index, status in ret_dict.items():
+            if int(status) == 1:
+                pass
+                #self._sfp_list[int(index)].check_sfp_optoe_type()
+        return ret_dict
+
+    def get_sfp(self, index):
+        """
+        Retrieves sfp represented by (1-based) index <index>
+
+        Args:
+            index: An integer, the index (1-based) of the sfp to retrieve.
+            The index should be the sequence of a physical port in a chassis,
+            starting from 1.
+            For example, 1 for Ethernet0, 2 for Ethernet4 and so on.
+
+        Returns:
+            An object derived from SfpBase representing the specified sfp
+        """
+        sfp = None
+
+        try:
+            # The index will start from 1
+            # sfputil already convert to physical port index according to config
+            sfp = self._sfp_list[index]
```

Comment:
Please check this once again. Generally, the index passed down from 'xcvrd' to chassis.py is  1 based. So, to get the required sfp obj from _sfp_list, you need to do 'index-1'.

Reply:
> Please check this once again. Generally, the index passed down from 'xcvrd' to chassis.py is 1 based. So, to get the required sfp obj from _sfp_list, you need to do 'index-1'.

I take back my comment. I just checked your port_config.ini and that is using 0-based indexing. So no requirement of 'index-1'. However, this function is not required then as the parent class 'chassis_base.py' provide the same definition.

 https://github.com/sonic-net/sonic-platform-common/blob/8c0398eb911ee96967c3a0e1b7fb3ba9167451e1/sonic_platform_base/chassis_base.py#L477

Reply:
Yes, you're right, the definition is duplicated, will update it.